src/python/pants/backend/python/macros/poetry_requirements_test.py

```diff
@@ -81,7 +82,44 @@ def test_max_tilde(test, exp) -> None:
     ],
 )
 def test_handle_str(test, exp) -> None:
-    assert parse_str_version("foo", test, "") == f"foo {exp}"
+    assert parse_str_version("foo", test, "", "") == f"foo {exp}"
+
+
+def test_add_markers() -> None:
+    # case with just mark
+
+    attr_mark = {"markers": "platform_python_implementation == 'CPython'"}
+    assert (
+        add_markers("foo==1.0.0", attr_mark, "somepath")
+        == "foo==1.0.0;platform_python_implementation == 'CPython'"
+    )
+
+    # case with adv mark
+    attr_mark_adv = {
+        "markers": "platform_python_implementation == 'CPython' or sys_platform == 'win32'"
+    }
+    assert (
+        add_markers("foo==1.0.0", attr_mark_adv, "somepath")
+        == "foo==1.0.0;(platform_python_implementation == 'CPython' or sys_platform == 'win32')"
+    )
+    # case with basic mark basic py
+    attr_basic_both = {"python": "3.6", **attr_mark}
+    assert (
+        add_markers("foo==1.0.0", attr_basic_both, "somepath")
+        == "foo==1.0.0;platform_python_implementation == 'CPython' and python_version == '3.6'"
+    )
+    # case with adv py basic mark
```

Comment:
```suggestion
    # Case with advanced Python constraint and basic marker.
```

Reply:
Oops, yeah, I meant to delete those (initially just for my own organization)...is it worth writing them all out for future editors?

Reply:
Feel free to delete. It's helpful that you already separate each edge case with a newline, and imo not totally necessary for a comment.